### PR TITLE
feature/environment

### PR DESCRIPTION
**Pull request by vancouver29**
the structure of Object Invironment is like this ?

### CODE REVIEW

File: src/physics/types.rs [hunk 1]
```diff
@@ -142,6 +142,26 @@ impl Field {
     }
 }
 
+
+// Invironment ///////////////////////////////////////////////////////////////
+
+pub struct Invironment {
```
Comment (johnxnguyen) [inline src/physics/types.rs]: There is a typo: Environment instead of Invironment.

File: src/physics/types.rs [hunk 2]
```diff
@@ -142,6 +142,26 @@ impl Field {
...
+
+impl Invironment {
+    pub fn update(&mut self) {
+
+        // update each body
+        for (id, field) in self.fields.iter_mut() {
+            
+        }
+    }
+
```
Comment (johnxnguyen) [inline src/physics/types.rs]: Try to keep the style consistent with the rest of the file. You should delete empty lines after method bodies. The same goes for before and after struct properties.

File: src/physics/types.rs [hunk 3]
```diff
@@ -142,6 +142,26 @@ impl Field {
...
+
+}
```
Comment (johnxnguyen) [inline src/physics/types.rs:159]: Also please add a new line between the implementation and the test header

**Review by johnxnguyen — CHANGES_REQUESTED** [review]

File: src/physics/types.rs [hunk 4]
```diff
@@ -142,6 +142,26 @@ impl Field {
     }
 }
 
+
+// Invironment ///////////////////////////////////////////////////////////////
+
+pub struct Invironment {
+
+    pub bodies: HashMap<u32, Body>,
+    pub fields: HashMap<u32, Field>,
```
Comment (johnxnguyen) [inline src/physics/types.rs]: I would suggest to store the fields in a `Vec`, rather than a `HashMap`, there's no reason to keep in in a `HashMap`.

**Review by johnxnguyen — CHANGES_REQUESTED** [review]

File: src/physics/types.rs (hunk 4 above)
Comment (vancouver29) [reply src/physics/types.rs]: how about this : 
`pub fields: Vec<Field>, ` ?

File: src/physics/types.rs [hunk 5]
```diff
@@ -142,6 +142,29 @@ impl Field {
     }
 }
 
+
+
+// Invironment ///////////////////////////////////////////////////////////////
```
Comment (johnxnguyen) [inline src/physics/types.rs]: Still a typo here.

File: src/physics/types.rs [hunk 6]
```diff
@@ -142,6 +142,29 @@ impl Field {
...
+pub struct Environment {
+    pub bodies: HashMap<u32, Body>,
+    pub fields: Vec<Field>,
+}
+
+
+impl Environment {
+    pub fn update(&mut self) {
+
+        // update each field
```
Comment (johnxnguyen) [inline src/physics/types.rs:154]: There are still some empty lines here.

**Review by johnxnguyen — DISMISSED** [review]
There are still some styling issues but otherwise looks good 👍

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**